test/models/manifest_test.rb

```diff
@@ -0,0 +1,86 @@
+require 'test_helper'
+
+class ManifestTest < ActiveSupport::TestCase
+
+  @@hash1 = { :name => "brian", :description => "ginger", :role => "developer" }
+  @@hash2 = { "name" => "jeremy", "description" => nil, "role" => "monsignor", "url" => nil }
+  @@hash3 = { name: "eric", description: "senior developer", role: "visionary" }
+  @@hash4 = { name: nil, description: nil, role: nil, url: "http://127.0.0.1" }
+
+  test "load a valid manifest" do
+    manifest = Manifest.load("test/fixtures/files/manifest_valid")
+    assert_instance_of Manifest, manifest, "Not a Manifest Object"
+  end
+
+  test "load an invalid manifest" do
+    manifest = Manifest.load("test/fixtures/files/manifest_invalid")
+    assert_instance_of InvalidManifest, manifest, "Not an InvalidManifest Object"
```

Comment:
In other words, outside users shouldn't need to know that "ManifestMissing and "InvalidManifest" classes actually exist. That is simply a Manifest implementation detail. The interface is `Manifest#valid?`

BTW I just realized that in accordance with this fact, we should probably move `attr_reader :exception` to the Manifest superclass. It can remain nil. That way the interface is the same. That is my bad.